### DISCUSSION TITLE
Why StringRelatedField is read_only?

### DISCUSSION THREAD

**Discussion by weber-s** [encode/django-rest-framework#8203] [Question & Answer] [answered]
Hello !

I wonder why a `StringRelatedField` is read only.

```py

class StringRelatedField(RelatedField):
    """
    A read only field that represents its targets using their
    plain string representation.
    """

    def __init__(self, **kwargs):
        kwargs['read_only'] = True
        super().__init__(**kwargs)

    def to_representation(self, value):
        return str(value)
```
but why? I needed to empty one StringRelatedField, and the solution was to go back to a ForeignKeyRelatedField, which ok for the update, but "ugly" for the get...

**Comment by lovelydinosaur** [comment, marked as answer]
Because there's no way to take a string, and lookup which instance would return that string.

**Reply by weber-s** [reply]
Ok, make sens for the lookup... Didn't think about that since I only wanted to delete it.
Thanks!
